src/DB.php

```diff
@@ -758,6 +766,7 @@ public static function insertMessageRequest(Message &$message)
             $sth->bindParam(':message_id', $message_id, \PDO::PARAM_INT);
             $sth->bindParam(':user_id', $from_id, \PDO::PARAM_INT);
             $sth->bindParam(':date', $date, \PDO::PARAM_STR);
+            $sth->bindParam(':edit_date', $edit_date, \PDO::PARAM_STR);
```

Comment:
I believe current CallbackqueryCommand is good example.

As per documentation https://core.telegram.org/bots/api#callbackquery callback query will either have Message object (messages posted by bot) [case 1] or inline_message_id (inline results posted by user via bot) [case 2]

Case 1:
- imagine /settings command that uses inlline buttons to navigate inside same message via editing it
- if this message is not in database it should be inserted to ```message``` table (ideally user's first interaction will insert the first state of the message - original message that was sent by the bot)
- every time message is edited (user navigating the settings buttons) either update the message in ```message``` table or add edited message to ```edited_message``` (second solution makes more sense and can provide edit history so people could see in their project in what order people are navigating settings therefore knowing which options catch bigger interest)

Case 2: (does not need any code changes)
- imagine the bot is providing simple images from inline search
- user posts result from the inline search with thumb up and down buttons
- clicking those buttons changes numbers next to each voting option (thumb up/down) which results in the message being edited by user (edited_message Update object is sent)

I'm currently using code solution that adds another query to be executed with each callback query, not an ideal solution I must say:

```php
public static function insertCallbackQueryRequest(CallbackQuery &$callback_query)
    {
        if (!self::isDbConnected()) {
            return false;
        }

        try {
            $mysql_query = 'INSERT IGNORE INTO `' . TB_CALLBACK_QUERY . '`
                (
                `id`, `user_id`, `chat_id`, `message_id`, `inline_message_id`, `data`, `created_at`
                )
                VALUES (
                :callback_query_id, :user_id, :chat_id, :message_id, :inline_message_id, :data, :created_at
                )';

            $sth_insert_callback_query = self::$pdo->prepare($mysql_query);

            $date = self::getTimestamp(time());

            $callback_query_id = $callback_query->getId();
            $from = $callback_query->getFrom();
            $user_id = null;
            if (is_object($from)) {
                $user_id = $from->getId();
                self::insertUser($from, $date);
            }

            $message = $callback_query->getMessage();
            $chat_id = null;
            $message_id = null;
            if ($message) {
                $chat_id = $message->getChat()->getId();
                $message_id = $message->getMessageId();

                $sth = self::$pdo->prepare('SELECT * FROM `' . TB_MESSAGE . '`
                    WHERE `id` = ' . $message_id . ' AND `chat_id` = ' . $chat_id
                );
                $result = $sth->execute();

                if ($result) {
                    if ($sth->rowCount() > 0) {
                        self::insertEditedMessageRequest($message);
                    } else {
                        self::insertMessageRequest($message);
                    }
                }
            }

            $inline_message_id = $callback_query->getInlineMessageId();
            $data = $callback_query->getData();

            $sth_insert_callback_query->bindParam(':callback_query_id', $callback_query_id, \PDO::PARAM_INT);
            $sth_insert_callback_query->bindParam(':user_id', $user_id, \PDO::PARAM_INT);
            $sth_insert_callback_query->bindParam(':chat_id', $chat_id, \PDO::PARAM_INT);
            $sth_insert_callback_query->bindParam(':message_id', $message_id, \PDO::PARAM_INT);
            $sth_insert_callback_query->bindParam(':inline_message_id', $inline_message_id, \PDO::PARAM_STR);
            $sth_insert_callback_query->bindParam(':data', $data, \PDO::PARAM_STR);
            $sth_insert_callback_query->bindParam(':created_at', $date, \PDO::PARAM_STR);

            return $sth_insert_callback_query->execute();
        } catch (PDOException $e) {
            throw new TelegramException($e->getMessage());
        }
    }
```

I basically check if the original message exists in ```message``` table and if not insert it there, otherwise insert into ```edited_message``` and reference original.

This code is a replacement for my 'ON DUPLICATE' solution in ```insertMessageRequest()``` commit.